libraries/lib-ipc/internal/socket_guard.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "ipc-types.h"
+
+class socket_guard final
```

Comment:
So I read the class to understand that it's a move-only object for RAII cleanup when a socket is opened.

Comment that before the class.  Perhaps too, "unique_socket_guard" would be a name that suggests that it is move-only.